packages/npm-dependencies-validation/src/depIssues.ts

```diff
@@ -0,0 +1,30 @@
+import { dirname } from "path";
+import { NpmLsResult } from "./types";
+import { isCurrentlySupported, isPathExist } from "./utils/packageJsonUtil";
+import { invokeNPMCommand } from "./utils/npmUtil";
+
+// ls --depth=0 shows only top-level dependencies
+const LS_ARGS: string[] = ["ls", "--depth=0"];
+
+export async function findDependencyIssues(
+  absPackageJsonPath: string
+): Promise<NpmLsResult> {
+  const packageJsonExists = await isPathExist(absPackageJsonPath);
+  if (!packageJsonExists) return { problems: [] };
+
+  const currentlySupported = await isCurrentlySupported(absPackageJsonPath);
+  if (!currentlySupported) return { problems: [] };
+
+  // dependencies issues and extraneous modules
+  const commandConfig = {
+    commandArgs: [...LS_ARGS],
+    cwd: dirname(absPackageJsonPath),
+  };
+  const npmLsResult = await invokeNPMCommand<NpmLsResult>(commandConfig);
+  const problemsLength = npmLsResult.problems?.length || 0;
+  if (problemsLength != 0) return npmLsResult;
```

Comment:
```typescript
const problemsLength = npmLsResult.problems?.length || 0;
  if (problemsLength != 0) return npmLsResult;
```

This looks a bit convoluted for the purpose of checking `isEmpty(npmLsResult.problems)`
Maybe you can replace it with `isEmpty` check?

Reply:
also "!=" should be "!=="

Reply:
```typescript
return npmLsResult;
```

This early return has a-lot of meaning.
We are ignoring devDeps and extraneous dependencies **if** there are other issues.
So an end user would have to perform the "npm install" **twice** in some edge cases.
And could be notified of issues twice in one flow.

This assumption / behavior needs to be :
- Well documented in the source code
- Synced with Product
- Synced with head arch